Ascend.java

```diff
@@ -2,6 +2,14 @@
 // and prints them in increasing order.
 public class Ascend {
 	public static void main(String[] args) {
-		// Replace this comment with your code
+		int lim = Integer.parseInt(args[0]);
+		int a = ((int)Math.random()*lim);
+		int b = (int)(Math.random()*lim);
+		int c = (int)(Math.random()*lim);
+		System.out.println(a+" "+b+" "+c);
+		int max = Math.max(a,Math.max(b,c));
+		int min = Math.min(a,Math.min(b,c));
+		int mid = Math.min(b,Math.max(a,c));

```

Comment:
Incorrect calc for mid
consider 
a = 3, b = 1, c = 2. 
according to your code 
mid = 1; since Math.min(1, Math.max(3, 2)) --> Math.min(1, 3) --> 1 

There are 2 possible solutions
1. `int mid = a + b + c - max - min;`
2. 
```
int midSuspect1 = Math.max(a, b);
int midSuspect2 = Math.max(a, c);
int midSuspect3 = Math.max(b, c);
int mid = Math.min(midSuspect3, Math.min(midSuspect1, midSuspect2));
```